pandas/core/frame.py

```diff
@@ -10435,7 +10452,34 @@ def apply(
         0  1  2
         1  1  2
         2  1  2
-        """
+
+        Advanced users can speed up their code by using a Just-in-time (JIT) compiler
+        with ``apply``. The main JIT compilers available for pandas are Numba and Bodo.
+        In general, JIT compilation is only possible when the function passed to
+        ``apply`` has type stability (variables in the function do not change their
+        type during the execution).
+
+        >>> import bodo
+        >>> df.apply(lambda x: x.A + x.B, axis=1, jit=bodo.jit(parallel=True))
+
+        Note that JIT compilation is only recommended for functions that take a
+        significant amount of time to run. Fast functions are unlikely to run faster
+        with JIT compilation.
+        """
+        if hasattr(jit, "__pandas_udf__"):
```

Comment:
what if jit is provided but doesnt have this attribute?

Reply:
what if engine or enging_kwargs are passed?

Reply:
Should error out I think.